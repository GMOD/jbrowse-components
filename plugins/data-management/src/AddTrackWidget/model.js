import { types } from 'mobx-state-tree'
import { ElementId } from '@jbrowse/core/util/types/mst'

export default pluginManager =>
  types.model('AddTrackModel', {
    id: ElementId,
    type: types.literal('AddTrackWidget'),
    view: types.safeReference(
      pluginManager.pluggableMstType('view', 'stateModel'),
    ),
  })
