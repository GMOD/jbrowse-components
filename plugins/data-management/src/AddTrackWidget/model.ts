/* eslint-disable @typescript-eslint/no-explicit-any */
import { types, Instance } from 'mobx-state-tree'
import PluginManager from '@jbrowse/core/PluginManager'
import { ElementId } from '@jbrowse/core/util/types/mst'
import {
  guessAdapter,
  guessTrackType,
  UNSUPPORTED,
} from '@jbrowse/core/util/tracks'

function isAbsoluteUrl(url: string) {
  try {
    // eslint-disable-next-line no-new
    new URL(url)
    return true
  } catch (error) {
    return url.startsWith('/')
  }
}

export default function f(pluginManager: PluginManager) {
  return types
    .model('AddTrackModel', {
      id: ElementId,
      type: types.literal('AddTrackWidget'),
      view: types.safeReference(
        pluginManager.pluggableMstType('view', 'stateModel'),
      ),
    })
    .volatile(() => ({
      trackSource: 'fromFile',
      trackData: { uri: '' } as any,
      indexTrackData: { uri: '' } as any,

      // alts
      altAssemblyName: '',
      altTrackName: '',
      altTrackType: '',

      altTrackAdapter: {} as any,
    }))
    .actions(self => ({
      setTrackAdapter(obj: any) {
        self.altTrackAdapter = obj
      },
      setTrackSource(str: string) {
        self.trackSource = str
      },
      setTrackData(obj: any) {
        self.trackData = obj
      },
      setIndexTrackData(obj: any) {
        self.indexTrackData = obj
      },
      setAssembly(str: string) {
        self.altAssemblyName = str
      },
      setTrackName(str: string) {
        self.altTrackName = str
      },
      setTrackType(str: string) {
        self.altTrackType = str
      },

      clearData() {
        self.trackSource = ''
        self.altTrackAdapter = {}
        self.indexTrackData = { uri: '' }
        self.trackData = { uri: '' }
        self.altTrackName = ''
        self.altTrackType = ''
        self.altAssemblyName = ''
        self.altTrackAdapter = {}
      },
    }))
    .views(self => ({
      get trackAdapter() {
        const { trackData, indexTrackData } = self
        if (trackData?.uri) {
          return guessAdapter(trackData.uri, 'uri', indexTrackData.uri)
        }

        if (trackData.localPath) {
          return guessAdapter(trackData.localPath, 'localPath')
        }
        return self.altTrackAdapter
      },

      get trackName() {
        const uri = self.trackData?.uri
        const localPath = self.trackData?.localPath
        return (
          self.altTrackName ||
          (uri ? uri.slice(uri.lastIndexOf('/') + 1) : null) ||
          (localPath ? localPath.slice(localPath.lastIndexOf('/') + 1) : null)
        )
      },

      get isFtp() {
        const { trackData, indexTrackData } = self
        return !!(
          (indexTrackData.uri && indexTrackData.uri.startsWith('ftp://')) ||
          (trackData.uri && trackData.uri.startsWith('ftp://'))
        )
      },

      get isRelativeUrl() {
        return !(
          (self.indexTrackData.uri && isAbsoluteUrl(self.indexTrackData.uri)) ||
          (self.trackData.uri && isAbsoluteUrl(self.trackData.uri))
        )
      },

      get wrongProtocol() {
        return (
          window.location.protocol === 'https:' &&
          ((self.indexTrackData.uri &&
            self.indexTrackData.uri.startsWith('http://')) ||
            self.trackData.uri.startsWith('http://'))
        )
      },

      get unsupported() {
        return this.trackAdapter.type === UNSUPPORTED
      },

      get assembly() {
        return self.altAssemblyName || self.view.assemblyNames?.[0]
      },

      get trackType() {
        return self.altTrackType || guessTrackType(this.trackAdapter.type)
      },
    }))
}

export type AddTrackStateModel = ReturnType<typeof f>
export type AddTrackModel = Instance<AddTrackStateModel>
