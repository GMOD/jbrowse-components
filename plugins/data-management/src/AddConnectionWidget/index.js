import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export { default as ReactComponent } from './components/AddConnectionWidget'
export { default as stateModel } from './model'
export const configSchema = ConfigurationSchema('AddConnectionWidget', {})
