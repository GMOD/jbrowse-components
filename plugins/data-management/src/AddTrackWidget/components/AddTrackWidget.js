import { getSession } from '@jbrowse/core/util'
import { getConf } from '@jbrowse/core/configuration'
import Button from '@material-ui/core/Button'
import Step from '@material-ui/core/Step'
import StepContent from '@material-ui/core/StepContent'
import StepLabel from '@material-ui/core/StepLabel'
import Stepper from '@material-ui/core/Stepper'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { observer, PropTypes as MobxPropTypes } from 'mobx-react'
import React, { useState } from 'react'
import ConfirmTrack from './ConfirmTrack'
import TrackSourceSelect from './TrackSourceSelect'

const useStyles = makeStyles(theme => ({
  root: {
    marginTop: theme.spacing(1),
  },
  stepper: {
    backgroundColor: theme.palette.background.default,
  },
  button: {
    marginTop: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  actionsContainer: {
    marginBottom: theme.spacing(2),
  },
  stepContent: {
    margin: theme.spacing(1),
  },
}))

const steps = ['Enter track data', 'Confirm track type']

function AddTrackWidget({ model }) {
  const [activeStep, setActiveStep] = useState(0)
  const [trackSource, setTrackSource] = useState('fromFile')
  const [trackData, setTrackData] = useState({ uri: '' })
  const [indexTrackData, setIndexTrackData] = useState({ uri: '' })
  const [trackName, setTrackName] = useState('')
  const [trackType, setTrackType] = useState('')
  const [trackAdapter, setTrackAdapter] = useState({})
  const [assembly, setAssembly] = useState(
    model.view.displayedRegions[0].assemblyName || '',
  )
  const classes = useStyles()
  const fileName = trackData.uri
    ? trackData.uri.slice(trackData.uri.lastIndexOf('/') + 1)
    : null

  const session = getSession(model)

  function getStepContent(step) {
    switch (step) {
      case 0:
        return (
          <TrackSourceSelect
            trackSource={trackSource}
            setTrackSource={setTrackSource}
            trackData={trackData}
            setTrackData={setTrackData}
            indexTrackData={indexTrackData}
            setIndexTrackData={setIndexTrackData}
          />
        )
      case 1:
        return (
          <ConfirmTrack
            fileName={fileName}
            session={session}
            indexTrackData={indexTrackData}
            trackData={trackData}
            trackName={trackName}
            setTrackName={setTrackName}
            trackType={trackType}
            setTrackType={setTrackType}
            trackAdapter={trackAdapter}
            setTrackAdapter={setTrackAdapter}
            assembly={assembly}
            setAssembly={setAssembly}
          />
        )
      default:
        return <Typography>Unknown step</Typography>
    }
  }

  function handleNext() {
    if (activeStep !== steps.length - 1) {
      setActiveStep(activeStep + 1)
      return
    }
    trackAdapter.features = trackData.config
    const trackId = `${trackName
      .toLowerCase()
      .replace(/ /g, '_')}-${Date.now()}`

    const assemblyInstance = session.assemblyManager.get(assembly)
    session.addTrackConf({
      trackId,
      type: trackType,
      name: trackName || fileName,
      assemblyNames: [assembly],
      adapter: {
        ...trackAdapter,
        sequenceAdapter: getConf(assemblyInstance, ['sequence', 'adapter']),
      },
    })
    if (model.view) {
      model.view.showTrack(trackId)
    } else {
      session.notify(
        'Open a new view, or use the track selector in an existing view, to view this track',
        'info',
      )
    }
    session.hideWidget(model)
  }

  function handleBack() {
    setActiveStep(activeStep - 1)
  }

  function isNextDisabled() {
    switch (activeStep) {
      case 0:
        return !(
          trackData.uri ||
          trackData.localPath ||
          trackData.blob ||
          trackData.config
        )
      case 1:
        return !(
          (trackName || fileName) &&
          trackType &&
          trackAdapter.type &&
          assembly
        )
      default:
        return true
    }
  }

  return (
    <div className={classes.root}>
      <Stepper
        className={classes.stepper}
        activeStep={activeStep}
        orientation="vertical"
      >
        {steps.map((label, idx) => (
          <Step key={label}>
            <StepLabel>{label}</StepLabel>
            <StepContent>
              {getStepContent(idx)}
              <div className={classes.actionsContainer}>
                <Button
                  disabled={activeStep === 0}
                  onClick={handleBack}
                  className={classes.button}
                >
                  Back
                </Button>
                <Button
                  disabled={isNextDisabled()}
                  variant="contained"
                  color="primary"
                  onClick={handleNext}
                  className={classes.button}
                  data-testid="addTrackNextButton"
                >
                  {activeStep === steps.length - 1 ? 'Add' : 'Next'}
                </Button>
              </div>
            </StepContent>
          </Step>
        ))}
      </Stepper>
    </div>
  )
}

AddTrackWidget.propTypes = {
  model: MobxPropTypes.observableObject.isRequired,
}

export default observer(AddTrackWidget)
