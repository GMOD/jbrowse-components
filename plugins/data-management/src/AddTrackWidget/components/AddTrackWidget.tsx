import { getSession } from '@jbrowse/core/util'
import { getConf } from '@jbrowse/core/configuration'
import Button from '@material-ui/core/Button'
import Step from '@material-ui/core/Step'
import StepContent from '@material-ui/core/StepContent'
import StepLabel from '@material-ui/core/StepLabel'
import Stepper from '@material-ui/core/Stepper'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { observer, PropTypes as MobxPropTypes } from 'mobx-react'
import React, { useState } from 'react'
import ConfirmTrack from './ConfirmTrack'
import TrackSourceSelect from './TrackSourceSelect'
import { AddTrackModel } from '../model'

const useStyles = makeStyles(theme => ({
  root: {
    marginTop: theme.spacing(1),
  },
  stepper: {
    backgroundColor: theme.palette.background.default,
  },
  button: {
    marginTop: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  actionsContainer: {
    marginBottom: theme.spacing(2),
  },
  stepContent: {
    margin: theme.spacing(1),
  },
}))

const steps = ['Enter track data', 'Confirm track type']

function AddTrackWidget({ model }: { model: AddTrackModel }) {
  const [activeStep, setActiveStep] = useState(0)
  const classes = useStyles()
  const session = getSession(model)
  const { assembly, trackAdapter, trackData, trackName, trackType } = model

  function getStepContent(step: number) {
    switch (step) {
      case 0:
        return <TrackSourceSelect model={model} />
      case 1:
        return <ConfirmTrack model={model} />
      default:
        return <Typography>Unknown step</Typography>
    }
  }

  function handleNext() {
    if (activeStep !== steps.length - 1) {
      setActiveStep(activeStep + 1)
      return
    }

    const trackId = `${trackName
      .toLowerCase()
      .replace(/ /g, '_')}-${Date.now()}`

    const assemblyInstance = session.assemblyManager.get(assembly)

    // @ts-ignore
    session.addTrackConf({
      trackId,
      type: trackType,
      name: trackName,
      assemblyNames: [assembly],
      adapter: {
        ...trackAdapter,
        sequenceAdapter: getConf(assemblyInstance, ['sequence', 'adapter']),
      },
    })
    if (model.view) {
      model.view.showTrack(trackId)
    } else {
      session.notify(
        'Open a new view, or use the track selector in an existing view, to view this track',
        'info',
      )
    }
    model.clearData()
    // @ts-ignore
    session.hideWidget(model)
  }

  function handleBack() {
    setActiveStep(activeStep - 1)
  }

  function isNextDisabled() {
    switch (activeStep) {
      case 0:
        return !(trackData.uri || trackData.localPath || trackData.blob)
      case 1:
        return !(trackName && trackType && trackAdapter.type && assembly)
      default:
        return true
    }
  }

  return (
    <div className={classes.root}>
      <Stepper
        className={classes.stepper}
        activeStep={activeStep}
        orientation="vertical"
      >
        {steps.map((label, idx) => (
          <Step key={label}>
            <StepLabel>{label}</StepLabel>
            <StepContent>
              {getStepContent(idx)}
              <div className={classes.actionsContainer}>
                <Button
                  disabled={activeStep === 0}
                  onClick={handleBack}
                  className={classes.button}
                >
                  Back
                </Button>
                <Button
                  disabled={isNextDisabled()}
                  variant="contained"
                  color="primary"
                  onClick={handleNext}
                  className={classes.button}
                  data-testid="addTrackNextButton"
                >
                  {activeStep === steps.length - 1 ? 'Add' : 'Next'}
                </Button>
              </div>
            </StepContent>
          </Step>
        ))}
      </Stepper>
    </div>
  )
}

AddTrackWidget.propTypes = {
  model: MobxPropTypes.observableObject.isRequired,
}

export default observer(AddTrackWidget)
