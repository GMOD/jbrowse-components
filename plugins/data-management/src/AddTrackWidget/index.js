import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export { default as ReactComponent } from './components/AddTrackWidget'
export { default as stateModelFactory } from './model'
export const configSchema = ConfigurationSchema('AddTrackWidget', {})
