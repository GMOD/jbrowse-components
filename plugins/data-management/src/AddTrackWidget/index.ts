import { ConfigurationSchema } from '@jbrowse/core/configuration'

export { default as stateModelFactory } from './model'
export const configSchema = ConfigurationSchema('AddTrackWidget', {})
