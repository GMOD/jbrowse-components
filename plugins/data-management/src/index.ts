import ConnectionType from '@gmod/jbrowse-core/pluggableElementTypes/ConnectionType'
import WidgetType from '@gmod/jbrowse-core/pluggableElementTypes/WidgetType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import PluginManager from '@gmod/jbrowse-core/PluginManager'
import {
  SessionWithWidgets,
  isAbstractMenuManager,
} from '@gmod/jbrowse-core/util'
import NoteAddIcon from '@material-ui/icons/NoteAdd'
import InputIcon from '@material-ui/icons/Input'
import {
  configSchema as ucscConfigSchema,
  modelFactory as ucscModelFactory,
} from './ucsc-trackhub'
import {
  ReactComponent as AddTrackReactComponent,
  stateModelFactory as AddTrackStateModelFactory,
  configSchema as AddTrackConfigSchema,
} from './AddTrackWidget'
import {
  ReactComponent as AddConnectionReactComponent,
  stateModel as AddConnectionStateModel,
  configSchema as AddConnectionConfigSchema,
} from './AddConnectionWidget'
import {
  ReactComponent as HierarchicalTrackSelectorReactComponent,
  stateModelFactory as HierarchicalTrackSelectorStateModelFactory,
  configSchema as HierarchicalTrackSelectorConfigSchema,
} from './HierarchicalTrackSelectorWidget'

export default class extends Plugin {
  name = 'DataManagementPlugin'

  install(pluginManager: PluginManager) {
    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'UCSCTrackHubConnection',
          configSchema: ucscConfigSchema,
          stateModel: ucscModelFactory(pluginManager),
          displayName: 'UCSC Track Hub',
          description: 'A track or assembly hub in the Track Hub format',
          url: '//genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Intro',
        }),
    )

    pluginManager.addWidgetType(() => {
      return new WidgetType({
        name: 'HierarchicalTrackSelectorWidget',
        heading: 'Available Tracks',
        configSchema: HierarchicalTrackSelectorConfigSchema,
        stateModel: HierarchicalTrackSelectorStateModelFactory(pluginManager),
        ReactComponent: HierarchicalTrackSelectorReactComponent,
      })
    })

    pluginManager.addWidgetType(() => {
      return new WidgetType({
        name: 'AddTrackWidget',
        heading: 'Add a track',
        configSchema: AddTrackConfigSchema,
        stateModel: AddTrackStateModelFactory(pluginManager),
        ReactComponent: AddTrackReactComponent,
      })
    })

    pluginManager.addWidgetType(() => {
      return new WidgetType({
        name: 'AddConnectionWidget',
        heading: 'Add a connection',
        configSchema: AddConnectionConfigSchema,
        stateModel: AddConnectionStateModel,
        ReactComponent: AddConnectionReactComponent,
      })
    })
  }

  configure(pluginManager: PluginManager) {
    if (isAbstractMenuManager(pluginManager.rootModel)) {
      pluginManager.rootModel.appendToMenu('File', {
        label: 'Open new track',
        icon: NoteAddIcon,
        onClick: (session: SessionWithWidgets) => {
          const widget = session.addWidget('AddTrackWidget', 'addTrackWidget')
          session.showWidget(widget)
        },
      })
      pluginManager.rootModel.appendToMenu('File', {
        label: 'Open new connection',
        icon: InputIcon,
        onClick: (session: SessionWithWidgets) => {
          const widget = session.addWidget(
            'AddConnectionWidget',
            'addConnectionWidget',
          )
          session.showWidget(widget)
        },
      })
    }
  }
}
