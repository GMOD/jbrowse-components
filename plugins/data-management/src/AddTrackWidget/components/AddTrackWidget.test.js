import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
} from '@testing-library/react'
import { createTestSession } from '@gmod/jbrowse-web/src/rootModel'
import AddTrackWidget from './AddTrackWidget'

describe('<AddTrackWidget />', () => {
  let session
  let model

  beforeAll(() => {
    session = createTestSession()
    session.addAssemblyConf({
      name: 'volMyt1',
      sequence: {
        trackId: 'ref0',
        type: 'ReferenceSequenceTrack',
        adapter: {
          type: 'FromConfigAdapter',
          features: [
            {
              refName: 'ctgA',
              uniqueId: 'firstId',
              start: 0,
              end: 1000,
              seq:
                'cattgttgcggagttgaacaACGGCATTAGGAACACTTCCGTCTCtcacttttatacgattatgattggttctttagccttggtttagattggtagtagtagcggcgctaatgctacctgaattgagaactcgagcgggggctaggcaaattctgattcagcctgacttctcttggaaccctgcccataaatcaaagggttagtgcggccaaaacgttggacaacggtattagaagaccaacctgaccaccaaaccgtcaattaaccggtatcttctcggaaacggcggttctctcctagatagcgatctgtggtctcaccatgcaatttaaacaggtgagtaaagattgctacaaatacgagactagctgtcaccagatgctgttcatctgttggctccttggtcgctccgttgtacccaggctactttgaaagagcgcagaatacttagacggtatcgatcatggtagcatagcattctgataacatgtatggagttcgaacatccgtctggggccggacggtccgtttgaggttggttgatctgggtgatagtcagcaagatagacgttagataacaaattaaaggattttaccttagattgcgactagtacaacggtacatcggtgattcgcgctctactagatcacgctatgggtaccataaacaaacggtggaccttctcaagctggttgacgcctcagcaacataggcttcctcctccacgcatctcagcataaaaggcttataaactgcttctttgtgccagagcaactcaattaagcccttggtaccgtgggcacgcattctgtcacggtgaccaactgttcatcctgaatcgccgaatgggactatttggtacaggaatcaagcggatggcactactgcagcttatttacgacggtattcttaaagtttttaagacaatgtatttcatgggtagttcggtttgttttattgctacacaggctcttgtagacgacctacttagcactacgg',
            },
          ],
        },
      },
    })
    session.addTrackConf({
      trackId: 'i3jUPmrgMOS',
      type: 'FilteringTrack',
      name: 'Filter Test',
      assemblyNames: ['volMyt1'],
      adapter: {
        type: 'FromConfigAdapter',
        features: [
          {
            uniqueId: 'one',
            refName: 'ctgA',
            start: 100,
            end: 101,
            type: 'foo',
            name: 'Boris',
            note: 'note for boris',
          },
          {
            uniqueId: 'two',
            refName: 'ctgA',
            start: 110,
            end: 111,
            type: 'bar',
            name: 'Theresa',
            note: 'note for theresa',
          },
          {
            uniqueId: 'three',
            refName: 'ctgA',
            start: 120,
            end: 121,
            type: 'baz',
            name: 'Nigel',
            note: 'note for nigel',
          },
          {
            uniqueId: 'four',
            refName: 'ctgA',
            start: 130,
            end: 131,
            type: 'quux',
            name: 'Geoffray',
            note: 'note for geoffray',
          },
        ],
      },
      renderer: {
        type: 'SvgFeatureRenderer',
        labels: {},
      },
      filterAttributes: ['type', 'start', 'end'],
    })
    const view = session.addView('LinearGenomeView', {
      displayedRegions: [
        {
          assemblyName: 'volMyt1',
          refName: 'ctgA',
          start: 0,
          end: 1000,
        },
      ],
    })
    model = session.addWidget('AddTrackWidget', 'addTrackWidget', {
      view: view.id,
    })
  })

  afterEach(cleanup)

  it('renders', () => {
    const { container } = render(<AddTrackWidget model={model} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('adds a track', async () => {
    const { getByTestId, getByText } = render(<AddTrackWidget model={model} />)
    expect(session.tracks.length).toBe(1)
    fireEvent.click(getByTestId('addTrackFromConfigRadio'))
    fireEvent.click(getByTestId('addTrackNextButton'))
    fireEvent.change(getByTestId('trackNameInput'), {
      target: { value: 'Test track name' },
    })
    const trackTypeSelect = getByTestId('trackTypeSelect')
    fireEvent.mouseDown(trackTypeSelect)
    const basicTrack = await waitForElement(() => getByText('BasicTrack'))
    fireEvent.click(basicTrack)
    const assemblyNameSelect = getByTestId('assemblyNameSelect')
    fireEvent.mouseDown(assemblyNameSelect)
    const volMyt1 = await waitForElement(() => getByText('volMyt1'))
    fireEvent.click(volMyt1)
    fireEvent.click(getByTestId('addTrackNextButton'))
    expect(session.tracks.length).toBe(2)
  })
})
