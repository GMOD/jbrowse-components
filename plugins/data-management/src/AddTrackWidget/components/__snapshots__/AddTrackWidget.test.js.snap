// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTrackWidget /> renders 1`] = `
<div
  class="makeStyles-root"
>
  <div
    class="MuiPaper-root MuiStepper-root MuiStepper-vertical makeStyles-stepper MuiPaper-elevation0"
  >
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Enter track data
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                class="makeStyles-root"
              >
                <fieldset
                  class="MuiFormControl-root"
                >
                  <div
                    aria-label="Data location"
                    class="MuiFormGroup-root"
                    role="radiogroup"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked MuiRadio-checked MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            checked=""
                            class="PrivateSwitchBase-input"
                            data-testid="addTrackFromFileRadio"
                            name="TrackSourceSelectRadioGroup"
                            type="radio"
                            value="fromFile"
                          />
                          <div
                            class="PrivateRadioButtonIcon-root PrivateRadioButtonIcon-checked"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        Enter data source
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            class="PrivateSwitchBase-input"
                            data-testid="addTrackFromConfigRadio"
                            name="TrackSourceSelectRadioGroup"
                            type="radio"
                            value="fromConfig"
                          />
                          <div
                            class="PrivateRadioButtonIcon-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        Enter data manually
                      </span>
                    </label>
                  </div>
                </fieldset>
                <div
                  class="MuiPaper-root makeStyles-paper MuiPaper-elevation1 MuiPaper-rounded"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink"
                    data-shrink="true"
                    for="callback-editor"
                  >
                    dataSourceLocation
                  </label>
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <div
                        aria-label="file or url picker"
                        class="MuiToggleButtonGroup-root"
                        role="group"
                      >
                        <button
                          aria-label="local file"
                          aria-pressed="false"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-disabled MuiToggleButton-sizeSmall MuiButtonBase-disabled"
                          disabled=""
                          tabindex="-1"
                          type="button"
                          value="file"
                        >
                          <span
                            class="MuiToggleButton-label"
                          >
                            File
                          </span>
                        </button>
                        <button
                          aria-label="url"
                          aria-pressed="true"
                          class="MuiButtonBase-root MuiToggleButton-root MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal MuiToggleButton-selected MuiToggleButton-sizeSmall"
                          tabindex="0"
                          type="button"
                          value="url"
                        >
                          <span
                            class="MuiToggleButton-label"
                          >
                            Url
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item"
                    >
                      <div
                        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
                      >
                        <div
                          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-marginDense MuiInput-marginDense"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiInput-input MuiInputBase-inputMarginDense MuiInput-inputMarginDense"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                  <p
                    class="MuiFormHelperText-root"
                  />
                </div>
              </div>
              <div
                class="makeStyles-actionsContainer"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button MuiButton-disabled MuiButtonBase-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Back
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button MuiButton-containedPrimary MuiButton-disabled MuiButtonBase-disabled"
                  data-testid="addTrackNextButton"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Next
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical MuiStepLabel-disabled"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Confirm track type
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        disabled=""
        icon="2"
      />
    </div>
  </div>
</div>
`;
